azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -129,6 +129,36 @@ def shared_conf(self):
     def remote_access(self):
         return self._remote_access
 
+    def fetch_agent_manifest(self, family_name, uris):
```

Comment:
The extensionsArtifact API requires the use of the x-ms-verify-from-artifacts-blob header for FastTrack goal states. I added these convenience functions to the GoalState class to make this easier.